knimin/tests/test_ag_search.py

```diff
@@ -64,7 +65,9 @@ def test_post_barcode(self):
                              field == 'ag_kit_barcode_id'):
                             self.assertNotIn(sample[field], response.body)
                         else:
-                            self.assertIn(sample[field], response.body)
+                            exp = xhtml_escape_recursive(sample[field])
```

Comment:
I'm a bit confused about this test. Basically, you have kit_id, then from that kit_id you loop over all the fields to check that they are in the response right? However, this should be a deterministic output, right? Not sure why the need to all these loops ...